crates/project/src/lsp_command/signature_help.rs

```diff
@@ -15,98 +16,109 @@ pub const SIGNATURE_HELP_HIGHLIGHT_CURRENT: MarkdownHighlight =
-            .or_else(|| help.signatures.first())?;
-
-        let str_for_join = ", ";
-        let parameter_length = signature_information
-            .parameters
-            .as_ref()
-            .map_or(0, |parameters| parameters.len());
-        let mut highlight_start = 0;
-        let (markdown, mut highlights): (Vec<_>, Vec<_>) = signature_information
-            .parameters
-            .as_ref()?
-            .iter()
-            .enumerate()
-            .map(|(i, parameter_information)| {
-                let label = match parameter_information.label.clone() {
-                    lsp::ParameterLabel::Simple(string) => string,
-                    lsp::ParameterLabel::LabelOffsets(offset) => signature_information
-                        .label
-                        .chars()
-                        .skip(offset[0] as usize)
-                        .take((offset[1] - offset[0]) as usize)
-                        .collect::<String>(),
-                };
-                let label_length = label.len();
-
-                let highlights = help.active_parameter.and_then(|active_parameter| {
-                    if i == active_parameter as usize {
-                        Some((
-                            highlight_start..(highlight_start + label_length),
-                            SIGNATURE_HELP_HIGHLIGHT_CURRENT,
-                        ))
-                    } else {
-                        None
+        let active_signature = help.active_signature? as usize;
+        let active_parameter = help.active_parameter.map(|n| n as usize);
+
+        let mut signature_helps = Vec::with_capacity(help.signatures.len());
+
+        for signature_information in help.signatures.clone() {
+            let str_for_join = ", ";
+            let parameter_length = signature_information
+                .parameters
+                .as_ref()
+                .map_or(0, |parameters| parameters.len());
+
+            let mut highlight: Option<(Range<usize>, MarkdownHighlight)> = None;
+            let mut highlight_start = 0;
+
+            let (markdown, description_markdown): (Vec<_>, Vec<_>) = signature_information
+                .parameters
+                .as_ref()?
+                .iter()
+                .enumerate()
+                .map(|(i, parameter_information)| {
+                    let label = match parameter_information.label.clone() {
+                        lsp::ParameterLabel::Simple(string) => string,
+                        lsp::ParameterLabel::LabelOffsets(offset) => signature_information
+                            .label
+                            .chars()
+                            .skip(offset[0] as usize)
+                            .take((offset[1] - offset[0]) as usize)
+                            .collect::<String>(),
+                    };
+                    let label_length = label.len();
+
+                    let documentation =
```

Comment:
This is wrong to consider any text as a markdown, as the latter implies specific char sequences to control its formatting, e.g. `*text*` rendered as plaintext vs *text* rendered as markdown.
So it's not correct to always do 
https://github.com/zed-industries/zed/blob/3666ba6a6ed9683c0907dcef73acc1e57206a042/crates/editor/src/signature_help.rs#L204-L209
as this PR does now.

We can use https://github.com/zed-industries/zed/blob/3666ba6a6ed9683c0907dcef73acc1e57206a042/crates/language/src/buffer.rs#L244 and render the text as a plain label in non-markdown cases.

Note that text could come from two places of the LSP type: 
* `lsp::Documentation::String`
* `lsp::Documentation::MarkupContent` for `MarkupKind::PlainText`